### PR TITLE
Adjust to Anitya related API changes

### PR DESCRIPTION
**Pull request by lbarcziova**
Related to packit/packit-service#2203
Related to packit/packit-service#2202

### CODE REVIEW

File: frontend/src/components/trigger/TriggerLink.tsx
Comment (Venefilyn) [inline frontend/src/components/trigger/TriggerLink.tsx:1]: We could do something like this to split it up and assert types. That way we require usage of `<TriggerSuffix>` to always provide a valid `repo_name` and `repo_namespace` combo or a valid `anitya_*` type. I checked a bunch of usages of `<TriggerLink>` and `<TriggerSuffix />` and it doesn't complain, but some stuff needs to be changed

<details>
<summary>suggested change in TriggerLink.tsx</summary>

```typescript
interface TriggerSuffixInterface {
  branch_name?: string | null;
  issue_id?: number | null;
  pr_id?: number | null;
  release?: string | null;
}

interface TriggerSuffixRepo extends TriggerSuffixInterface {
  repo_name: string;
  repo_namespace: string;
}

interface TriggerSuffixAnitya extends TriggerSuffixInterface {
  anitya_project_name: string;
  anitya_version: string;
  anitya_package: string;
}

type TriggerSuffixProps = {
  trigger: TriggerSuffixRepo | TriggerSuffixAnitya;
};

const TriggerSuffix: React.FC<TriggerSuffixProps> = ({ trigger }) => {
  // set suffix to be either PR ID or Branch Name depending on trigger
  let jobSuffix = "";

  if ("anitya_version" in trigger) {
    jobSuffix = `#version:${trigger.anitya_version}`;
    return (
      <>
        {trigger.anitya_project_name}/{jobSuffix}
      </>
    );
  } else if (trigger.pr_id) {
    jobSuffix = `#${trigger.pr_id}`;
  } else if (trigger.issue_id) {
    jobSuffix = `#${trigger.issue_id}`;
  } else if (trigger.branch_name) {
    jobSuffix = `:${trigger.branch_name}`;
  } else if (trigger.release) {
    jobSuffix = `#release:${trigger.release}`;
  }

  return (
    <>
      {trigger.repo_namespace}/{trigger.repo_name}
      {jobSuffix}
    </>
  );
};
```
</details>

And we need to change all `apiDefinitions.ts` `antiya_*` properties to be from `unknown | null` to `string | null`

**Review by Venefilyn — CHANGES_REQUESTED** [review]

File: frontend/src/components/trigger/TriggerLink.tsx [hunk 1]
```diff
@@ -67,6 +76,13 @@ const TriggerSuffix: React.FC<TriggerSuffixProps> = ({ trigger }) => {
     jobSuffix = `:${trigger.branch_name}`;
   } else if (trigger.release) {
     jobSuffix = `#release:${trigger.release}`;
+  } else if (trigger.anitya_version) {
+    jobSuffix = `#version:${trigger.anitya_version}`;
+    return (
+      <>
+        {trigger.anitya_project_name}/{jobSuffix}
+      </>
+    );
```
[the remark this answers is not in the document]
Comment (lbarcziova) [reply frontend/src/components/trigger/TriggerLink.tsx:85]: yes, that's right

File: frontend/src/components/trigger/TriggerLink.tsx
Comment (lbarcziova) [reply frontend/src/components/trigger/TriggerLink.tsx:1]: with your suggestion, the `("anitya_version" in trigger)` was always evaluated as `true`, so I changed it a little bit so locally it works, but I am not really sure if this is correct, PTAL :D

Comment (Venefilyn) [reply frontend/src/components/trigger/TriggerLink.tsx:1]: Okay figured it out. The suggestion you added was something I thought would work but evidently you need to define that somewhere, which makes it a bit more difficult. I did however find a way to get it to work correctly as far as I can tell

```
interface TriggerSuffixGitRepo extends TriggerSuffixInterface {
  repo_name: string;
  repo_namespace: string;
  anitya_project_name?: string | null;
  anitya_version?: string | null;
  anitya_package?: string | null;
}

interface TriggerSuffixAnityaProject extends TriggerSuffixInterface {
  anitya_project_name: string;
  anitya_version: string;
  anitya_package: string;
  repo_name?: string | null;
  repo_namespace?: string | null;
}

type TriggerSuffixProps = {
  trigger: TriggerSuffixGitRepo | TriggerSuffixAnityaProject;
};

const TriggerSuffix: React.FC<TriggerSuffixProps> = ({ trigger }) => {
  let jobSuffix = "";

  if (isAnityaTrigger(trigger)) {
    jobSuffix = `#version:${trigger.anitya_version}`;
    return (
      <>
        {trigger.anitya_project_name}/{jobSuffix}
      </>
    );
  } else if (trigger.pr_id) {
    jobSuffix = `#${trigger.pr_id}`;
  } else if (trigger.issue_id) {
    jobSuffix = `#${trigger.issue_id}`;
  } else if (trigger.branch_name) {
    jobSuffix = `:${trigger.branch_name}`;
  } else if (trigger.release) {
    jobSuffix = `#release:${trigger.release}`;
  }

  return (
    <>
      {trigger.repo_namespace}/{trigger.repo_name}
      {jobSuffix}
    </>
  );
};

function isAnityaTrigger(trigger: TriggerSuffixProps["trigger"]): trigger is TriggerSuffixAnityaProject {
  return typeof trigger.anitya_version === "string" &&
    trigger.anitya_version.length > 0
}
```

It has to do with the last bit in the code, which is a [type guard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)

Comment (lbarcziova) [reply frontend/src/components/trigger/TriggerLink.tsx:1]: thanks, added!

**Comment by lbarcziova** [comment]
regarding the icons, I was choosing between 

Snímka obrazovky z 2024-08-21 10-44-31
Snímka obrazovky z 2024-08-21 10-44-09

let me know which one you like more (even some other from [here](https://www.patternfly.org/design-foundations/icons/))

**Review by Venefilyn — APPROVED** [review]
LGTM, just the icon selection left then :D

**Comment by mfocko** [comment]
> regarding the icons, I was choosing between
> 
> Snímka obrazovky z 2024-08-21 10-44-31 Snímka obrazovky z 2024-08-21 10-44-09
> 
> @packit/the-packit-team , let me know which one you like more (even some other from [here](https://www.patternfly.org/design-foundations/icons/))

first :)

File: frontend/src/apiDefinitions.ts [hunk 2]
```diff
@@ -102,10 +102,10 @@ export interface CoprBuildPackage {
 
 // /api/copr-builds/$id
 export interface CoprBuild {
-  antiya_package: unknown | null;
```
Comment (Venefilyn) [inline frontend/src/apiDefinitions.ts:105]: Just saw I made a typo that you fixed here. Anitya is really bad for my dyslexia

**Comment by lachmanfrantisek** [comment]
Looks like the pipeline view does not like this change..;(

image

**Comment by lbarcziova** [comment]
@lachmanfrantisek will have a look into it today, thanks for letting me know.

**Comment by Venefilyn** [comment]
@lbarcziova I got back from PTO today and can also take a look if you want, just ping me if so

**Comment by lbarcziova** [comment]
@Venefilyn no problem, I also just got from the PTO, but opened #446 for this. Though if I checked correctly, the exception was not related to changes in this PR at all, it looks like just runs occasionally missing the trigger in the response.

**Pull request merged**